cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreCtx.java

```diff
@@ -689,14 +690,22 @@ public void setSeqMax(int seqMax) {
 	 */
 	private void setLengths() {
 		if (common_alg != null) {
-			if (common_alg.equals(AlgorithmID.AES_CCM_16_64_128)) {
-				iv_length = ivLength(common_alg);
-				id_length = 7;
-				key_length = common_alg.getKeySize() / 8; // 16;
-			} else {
+
+			switch (common_alg) {
+			case AES_CCM_16_64_128:
+			case AES_CCM_64_64_128:
+			case AES_CCM_16_128_128:
+			case AES_CCM_64_128_128:
+				iv_length = EncryptCommon.ivLength(common_alg);
```

Comment:
`ivLength` would return a `-1` for not supported algorithms (or maybe a exception after redesign). Checking for the returned value (or exception) would obsolete the repeated "switch(algo)".

Reply:
It's OK, if you want to keep this switch. I would replace it with `iv_length > 0`.

Reply:
I will update this one also. I was too focused on the EncryptCommon and missed this one.